### PR TITLE
Adding Support for multiple (long) restrictions for the same mode

### PR DESCRIPTION
**Pull request by kdiluca**
# Issue

We want to be able to support multiple restrictions for the same mode.  

## Tasklist

 - [x] Add tests
 - [ ] Review - you must request approval to merge any PR to master
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] Update relevant [documentation](docs/README.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by kdiluca** [comment]
@gknisely I have a ticket in for requesting assistance in creating more efficient queries from the HIVE tables to create test requests that demonstrate using the restrictions: https://github.com/mapbox/roads/issues/134

**Comment by PureW** [comment]
Could you break out the third-party rapid-json upgrade to a separate PR?

File: test/astar.cc [hunk 1]
```diff
@@ -1102,6 +1102,141 @@ void test_time_restricted_road_allowed_on_timedep() {
...
+
+// The following requests results in timedep astar during the non-restricted hours
+// and should be allowed
+void test_timed_no_access_restriction_1() {
+  auto response = timed_access_restriction("bicycle", "2018-05-13T19:14");
+  const auto& legs = response.trip().routes(0).legs();
+  const auto& directions = response.directions().routes(0).legs(0);
+  const auto& maneuvers_size = directions.maneuver_size();
+  if (maneuvers_size != 3) {
+    throw std::logic_error("This route should remain on Orchard St.");
```
Comment (PureW) [inline test/astar.cc]: This error message seems different from the assertion that there should be 3 maneuvers?

File: test/astar.cc [hunk 2]
```diff
@@ -1102,6 +1102,141 @@ void test_time_restricted_road_allowed_on_timedep() {
...
+  }
+}
+
+void test_timed_conditional_restriction_2() {
+  auto response = timed_conditional_restriction("auto", "2018-11-01T09:00");
+  const auto& legs = response.trip().routes(0).legs();
+  const auto& directions = response.directions().routes(0).legs(0);
+  const auto& maneuvers_size = directions.maneuver_size();
+  if (maneuvers_size > 3) {
+    throw std::logic_error("This route should turn L onto Dickinson Ave.");
```
Comment (PureW) [inline test/astar.cc]: Same comment, error message is different from the assertion of maneuver_size.

File: test/astar.cc [hunk 3]
```diff
@@ -1102,6 +1102,141 @@ void test_time_restricted_road_allowed_on_timedep() {
...
+
+// The following requests results in timedep astar during the restricted hours
+// and should be denied
+void test_timed_conditional_restriction_1() {
+  auto response = timed_conditional_restriction("auto", "2018-11-01T07:00");
+  const auto& legs = response.trip().routes(0).legs();
+  const auto& directions = response.directions().routes(0).legs(0);
+  const auto& maneuvers_size = directions.maneuver_size();
+  if (maneuvers_size > 3) {
+    throw std::logic_error("This route should turn L onto Dickinson Ave.");
```
Comment (PureW) [inline test/astar.cc]: Same comment, error message is different from the assertion of maneuver_size.

File: test/astar.cc [hunk 4]
```diff
@@ -1102,6 +1102,141 @@ void test_time_restricted_road_allowed_on_timedep() {
...
+  }
+}
+
+void test_timed_no_conditional_restriction_2() {
+  auto response = timed_conditional_restriction("auto", "2018-11-01T10:00");
+  const auto& legs = response.trip().routes(0).legs();
+  const auto& directions = response.directions().routes(0).legs(0);
+  const auto& maneuvers_size = directions.maneuver_size();
+  if (maneuvers_size != 3) {
+    throw std::logic_error("This route should turn R onto Dickinson Ave.");
```
Comment (PureW) [inline test/astar.cc]: Same comment, error message is different from the assertion of maneuver_size.

File: test/astar.cc [hunk 5]
```diff
@@ -1102,6 +1102,141 @@ void test_time_restricted_road_allowed_on_timedep() {
...
+
+// The following requests results in timedep astar during the non-restricted hours
+// and should be allowed
+void test_timed_no_conditional_restriction_1() {
+  auto response = timed_conditional_restriction("auto", "2018-11-01T06:30");
+  const auto& legs = response.trip().routes(0).legs();
+  const auto& directions = response.directions().routes(0).legs(0);
+  const auto& maneuvers_size = directions.maneuver_size();
+  if (maneuvers_size != 3) {
+    throw std::logic_error("This route should turn R onto Dickinson Ave.");
```
Comment (PureW) [inline test/astar.cc]: Same comment, error message is different from the assertion of maneuver_size.

**Pull request merged**